src/FetchLoader.js

```diff
@@ -58,48 +45,51 @@ class FetchLoader extends React.Component {
           <p>Loading, please wait...</p>
         </div> :
         data.results && data.results.length > 0 ?
-          <FacetedSearchContainer {...data}
-                                  ResultElementClass={ResultElementClass}
-                                  resultsMessage={resultsMessage}/> :
+          <FacetedSearchContainer
+            {...data}
+            ResultElementClass={ResultElementClass}
+            resultsMessage={resultsMessage}/> :
           <div className={`row column`}>
             <p>{noResultsMessage}</p>
             { data.reason && <p>{data.reason}</p> }
           </div>
     )
   }
 
-  _fetchAndSetState(host, resource) {
-    this.setState({ loading: true })
+  async componentDidUpdate(prevProps, prevState) {
+    if (this.state.data === null && this.state.error === null) {
```

Comment:
I wonder that `componentDidUpdate` method should check if `prevState` is `null` instead of `state` to initialize the component?

Reply:
No, you really want it to be `this.state`. If the previous state had fetched data, `prevState.data === null` would always be false and new data wouldn’t be requested. `prevProps` and `prevState` are passed in order to implement more complex behaviour, but they’re not always needed.

Reply:
As far as I know, `componentDidUpdate` should have a condition to check the equality between the pair of`prevState` and `state` or `prevProps` and `props` to execute the following codes, instead of checking the current `state`.

Reply:
Again, according to [React‘s guide](https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html#fetching-external-data-when-props-change):
>>> The recommended upgrade path for this component is to move data updates into `componentDidUpdate`.

I guess that since `state` can change in `getDerivedStateFromProps`, and that comes before `componentDidUpdate`, the steps are:
1. Mutate `state` from `props`  in `getDerivedStateFromProps`: if the new `state` is the same as before return `null` and `componentDidUpdate` isn’t called.
2. If the `state` is differente, do async fetch in `componentDidUpdate` based on the new `state`.